lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs

```diff
@@ -234,6 +235,8 @@ const rule = (primary, secondaryOptions, context) => {
 			}
 
 			for (const longhand of longhandProps) {
+				if (secondaryOptions?.ignoreLonghands?.includes(longhand)) continue;
```

Comment:
[suggest] How about defining a local variable outside the loop like this?

```js
		if (!validOptions) {
			return;
		}

		/** @type {string[]} */
		const ignoreLonghands = secondaryOptions?.ignoreLonghands ?? [];
```

Reply:
I wish we didn't have to cast it.
Can you take a look at #7608?

Reply:
The idea of `meta.schema` sounds interesting, but casting is necessary for now, right?